inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/rocks/inspectit/oce/core/config/DirectoryPropertySource.java

```diff
@@ -0,0 +1,146 @@
+package rocks.inspectit.oce.core.rocks.inspectit.oce.core.config;
+
+import lombok.Getter;
+import lombok.experimental.var;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.springframework.core.env.EnumerablePropertySource;
+import org.springframework.core.env.MutablePropertySources;
+import org.springframework.core.env.PropertiesPropertySource;
+import org.springframework.core.env.PropertySource;
+import org.springframework.core.io.FileSystemResource;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.stream.Stream;
+
+/**
+ * Special Properties source which adds all .properties and .yml/.yaml files form a directory as property sources.
+ * No subdirectories are parsed!
+ * The files are parsed in alphabetical order where the configuration from the first found file wins.
+ * This is done by adding the DirectoryPropertySource as marker which is followed by all contained files as PropertySources.
+ * {@link org.springframework.core.env.CompositePropertySource} was not used as this causes issues with the overwriting of Lists.
+ *
+ * @author Jonas Kunz
+ */
+@Slf4j
+public class DirectoryPropertySource extends EnumerablePropertySource<Void> {
+
+
+    private static final List<String> PROPERTIES_ENDINGS = Arrays.asList(".properties");
+    private static final List<String> YAML_ENDINGS = Arrays.asList(".yml", ".yaml");
+
+    @Getter
+    private Path rootDir;
+
+    /**
+     * stores all contained PropertySources (=one for each contained file)
+     */
+    private final List<PropertySource<?>> ownedFilePropertySources = new ArrayList<>();
+
+    private MutablePropertySources container;
+
+    /**
+     * Factory method.
+     * Creates a DirectoryPropertySource and places it as last source in the given container.
+     * @param name the name of the DirectoryPropertySource to create. This is used as basename for the PropertySources generated for each contained file.
+     * @param rootDir the directory to parse for config files
+     * @param container the container to append the PropertySources to
+     * @return
+     */
+    public static DirectoryPropertySource addLast(String name, Path rootDir, MutablePropertySources container) {
+        val dps = new DirectoryPropertySource(name,rootDir,container);
```

Comment:
is the save actions activated for you? this line is not formatted, can you check quickly, please..